code/parachain/runtime/picasso/src/lib.rs

```diff
@@ -858,7 +859,34 @@ impl_runtime_apis! {
 		}
 
 		fn list_assets() -> Vec<Asset<Balance, ForeignAssetId>> {
-			CurrencyId::list_assets()
+			// Hardcoded assets
+			let assets = CurrencyId::list_assets().into_iter().map(|mut asset| {
+				// Add hardcoded ratio and ED for well known assets
+				asset.ratio = WellKnownForeignToNativePriceConverter::get_ratio(CurrencyId(asset.id));
+				asset.existential_deposit = multi_existential_deposits::<AssetsRegistry>(&asset.id.into());
+				asset
+			}).collect::<Vec<_>>();
+
+			// Assets from the assets-registry pallet
+			let foreign_assets = assets_registry::Pallet::<Runtime>::get_foreign_assets_list();
+
+			// Override asset data for hardcoded assets that have been manually updated, and append
+			// new assets without duplication
+			foreign_assets.into_iter().fold(assets, |mut acc, mut foreign_asset| {
+				if let Some(i) = acc.iter().position(|asset_i| asset_i.id == foreign_asset.id) {
+					// Assets that have been updated
+					if let Some(asset) = acc.get_mut(i) {
+						// Update asset with data from assets-registry
+						asset.decimals = foreign_asset.decimals;
+						asset.foreign_id = foreign_asset.foreign_id.clone();
+						asset.ratio = foreign_asset.ratio;
```

Comment:
I thought we agreed to not overwrite well-known assets with what's inside Asset Registry?
Regardless, I think this assignment could be simplified. See:
https://doc.rust-lang.org/core/option/enum.Option.html#method.replace
https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/crowdloan-rewards/src/lib.rs#L458-L462

Reply:
yes, Vim had the same question when I added this to Dali (this PR is only copying the RPC from Dali) https://github.com/ComposableFi/composable/pull/2312#discussion_r1025806521

As we are not going to override it, this part won't have any effect. But if at some point we want to override one of the well known assets, it will use the updated information.

Regarding the `.replace` method, looks neat! I'll change it to use that.